dohproxy/client_protocol.py

```diff
@@ -145,3 +165,14 @@ async def make_request(self, addr, dnsq):
         # Read response trailers
         trailers = await client.recv_trailers(stream_id)
         self.logger.debug('Response trailers: {}'.format(trailers))
+
+    async def resolve(self, addr, dnsq):
+        qid = dnsq.id
+        dnsq.id = 0
+        dnsclient = DNSClient(self.args.passthrough_resolver,
+                              self.args.passthrough_port,
+                              logger=self.logger)
+        dnsr = await dnsclient.query(dnsq, addr)
+        dnsr.id = qid
```

Comment:
I don't think you need to care for the `qid` here. `DNSClient._try_query` does that for you (e.g replace with a random one when sending out, setting it back to original value when receiving an answer.